src/containers/articles/Article.js

```diff
@@ -5,8 +5,14 @@ import { getArticle, deleteArticle, toggleLike } from '../../actions/articleActi
 import Article from '../../components/articles/Article';
 
 class ArticlePage extends Component {
+  componentWillMount() {
+    const { articleData } = this.props; // eslint-disable-line
+    articleData.id = 0;
+  }
+
   componentDidMount() {
     const { articleData, getArticle, history } = this.props; // eslint-disable-line
+    articleData.id = 0;
```

Comment:
This logic does not make sense to me. 
You have already hard coded the value, it will always be 0 in this function.

Reply:
When you click on an article from say from the homepage as it is right now it will open the article and update the global state with the article data. When you open your profile and click on an article you wrote maybe just to view it, the former article is still in the state and shows for some time before the expected article shows up thus I have to set the id to 0 to set the loader  to visible and prevent this instance where a formerly loaded article shows up before the expected one is completely fetched

Reply:
cool